### DISCUSSION TITLE
Password "brownout"

### DISCUSSION THREAD

**Discussion by sglanger** [community/community#24714] [Apps, API and Webhooks]
<p>this AM i pushed some new code via “git push” over ssh and it worked fine… 10 minutes ago I got the below. The s link seems old and not relevant to ssh. HOw do I setup this token?</p>
<p>/***<br>
remote: Password authentication is temporarily disabled as part of a brownout. Please use a personal access token instead.<br>
remote: Please see <a href="https://github.blog/2020-07-30-token-authentication-requirements-for-api-and-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for API and Git operations | The GitHub Blog</a> for more information.<br>
**/</p>

**Comment by airtower-luna** [comment]
<p>You’re right that that shouldn’t apply when using SSH, tokens are for Git over HTTPS. I wonder if that repository happens to use HTTPS, maybe with a cached password, for some reason? You can use <code>git remote show origin</code> to check (assuming your remote is named <code>origin</code>, which is the default for where you cloned from).</p>

**Comment by sglanger** [comment]
<p>nope, does not use https<br>
and I see others are having issue now even with the token</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/sglanger">@sglanger</a> :wave: and <a class="mention" href="/airtower-luna">@airtower-luna</a> :wave:</p>
<p>From my understanding of the blog post referenced, the concern is that password authentication for Git operations over SSH should have worked during <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/#july-28-2021">this scheduled brownout</a>. Is that right?</p>
<p>I’m not sure I understand the same, and would have expected the <code>git pushed</code> to fail, if using pw auth.</p>
<p>For how to setup your SSH key, docs begin here:</p>
<p><a href="https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh</a></p>
<p>Though it was mentioned that other users are/were having issues <em>with</em> a valid SSH token, and receiving the same message back? Could you link us to an example thread?</p>

**Comment by sglanger** [comment]
<p>other examples</p><aside class="quote" data-post="1" data-topic="192601">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/all-3rd-party-connection-to-github-failed-with-403/192601">All 3rd party connection to Github failed with 403</a> <a class="badge-wrapper  bullet" href="/c/github-ecosystem/37"><span class="badge-category-bg" style="background-color: #28a745;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and support using GitHub’s REST, and GraphQL APIs, building Applications and OAuth Apps, the GitHub Marketplace, and third party integrations.">GitHub Ecosystem</span></a>
  </div>
  <blockquote>
    I use sourcetree and connect to github via basic auth (already use personal access token as password) but today I can’t pull from origin with error 403 
but I can pull form git CLI with same token 
and I use webhook in my repository, but today it got same 403 error too 
Can someone has any suggestion for me
  </blockquote>
</aside>

<p>But never mind, today it works again</p>

**Comment by sglanger** [comment]
<p>PS: Nethgato, you mentioned this was a “scheduled” brownout. Where was that announcement posted? I missed it, and it does not mention it was scheduled in the error I first posted.</p>

**Comment by adriennrichmond** [comment, marked as answer]
<p><a class="mention" href="/sglanger">@sglanger</a> The brownout dates and times were listed in our blog post:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener" title="08:28PM - 15 December 2020">The GitHub Blog – 15 Dec 20</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/09/security-1200-630.png?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" target="_blank" rel="noopener">Token authentication requirements for Git operations | The GitHub Blog</a></h3>

  <p>Beginning August 13th, 2021, we will no longer accept account passwords when authenticating Git operations on GitHub.com.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<p>
Sorry if the information has missed you, apologies for the inconvenience.</p>

**Comment by sglanger** [comment]
<p>Thx, it would have been useful to have that info in the error message from yesterday.  The blog link that was included in the error did not have the Brownout schedule in it.</p>

**Comment by nethgato** [comment]
<p><a class="mention" href="/sglanger">@sglanger</a> you’re right:</p>
<blockquote>
<p>it would have been useful to have that info in the error message from yesterday. The blog link that was included in the error did not have the Brownout schedule in it.</p>
</blockquote>
<p>I was disappointed to see that the blog post that did <em>not</em> include the specific time and date, but the initial plan. We’ve elevated that detail to our dev team to consider for the future.</p>
<p>For the auth failing with token based auth, that experience seems to be unrelated to the scheduled brown out for pw based auth.</p>

**Comment by sglanger** [comment]
<p>I’d be willing to bet that if you compare the number of accounts on github and compare to the number who have read that post, it will be a rather large difference.</p>

**Comment by usuargithub** [comment]
<aside class="quote group-github-employees" data-username="adriennrichmond" data-post="8" data-topic="192588">
<div class="title">
<div class="quote-controls"></div>
 adriennrichmond:</div>
<blockquote>
<p><a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a></p>
</blockquote>
</aside>
<p>Hello,<br>
did I miss something? That date is nine days ahead, but if I want to login in community, always have to post token.<br>
The correct date is 2020?</p>
<p>Regards</p>

**Comment by airtower-luna** [comment]
<p>The linked post is about Git operations, not logging in to GitHub or the community forum (here). Where to you have to post a token to log in here?</p>

**Comment by usuargithub** [comment]
<p>Hello,<br>
<a>Wird verarbeitet: Captura de pantalla de 2021-08-06 15-10-06.png …</a><br>
always happens.</p>
<p>Regards</p>

**Comment by airtower-luna** [comment]
<p>That looks like you tried to post a picture (screenshot, maybe?) but it didn’t work. Could you try again?</p>

**Comment by usuargithub** [comment]
<p>Hello,<br>
yes, it is a screenshot, but this is only one more difficulty for me because using a old browser-version.</p>
<p>Regards</p>

**Comment by usuargithub** [comment]
<p><a>Wird verarbeitet: Captura de pantalla de 2021-08-06 15-10-06.png …</a></p>

**Comment by usuargithub** [comment]
<p>Hello,<br>
doesn’t work:<br>
" Entschuldige, es gab einen Fehler beim Hochladen der Datei. Bitte versuche es noch einmal.</p>
<p>[OK]"</p>
<p>Regards</p>

**Comment by usuargithub** [comment]
<p>Hello,<br>
to login I have to do 2 way indent.<br>
Images-posting for mi is working again.</p>
<p>Regards</p>

**Comment by usuargithub** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/f/9f1338393f256bd8a3636025785934c6254d3054.png" data-download-href="/uploads/short-url/mHf5VHeuAd3SFdKSgcITCgt8Che.png?dl=1" title="Captura de pantalla de 2021-08-14 16-15-56" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/31926159/181086413-78bb6019-ea14-4721-ba95-058799290a35.png" alt="Captura de pantalla de 2021-08-14 16-15-56" data-base62-sha1="mHf5VHeuAd3SFdKSgcITCgt8Che" width="690" height="297" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/f/9f1338393f256bd8a3636025785934c6254d3054_2_690x297.png, https://aws1.discourse-cdn.com/github/original/3X/9/f/9f1338393f256bd8a3636025785934c6254d3054.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/f/9f1338393f256bd8a3636025785934c6254d3054.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/f/9f1338393f256bd8a3636025785934c6254d3054_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Captura de pantalla de 2021-08-14 16-15-56</span><span class="informations">1000×431 24 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Here (community) all working fine.</p>
<p>Regards</p>

**Comment by airtower-luna** [comment]
<p>That’s normal if you’re logging in without 2FA. You should have received the code by email. That has nothing to do with the PAT you need for Git operations over HTTPS, though.</p>
